internal/convert/automation_config.go

```diff
@@ -56,20 +55,34 @@ func FromAutomationConfig(in *opsmngr.AutomationConfig) (out []ClusterConfig) {
 	return
 }
 
-const keyLength = 500
+const (
+	keyLength = 500
+	cr        = "MONGODB-CR"
+	sha256    = "SCRAM-SHA-256"
+)
 
 func EnableMechanism(out *opsmngr.AutomationConfig, m []string) error {
-	out.Auth.DeploymentAuthMechanisms = append(out.Auth.DeploymentAuthMechanisms, m...)
-	out.Auth.AutoAuthMechanisms = append(out.Auth.AutoAuthMechanisms, m...)
 	out.Auth.Disabled = false
+	for _, v := range m {
+		if v != cr && v != sha256 {
+			return fmt.Errorf("unsupported mechanism %s", v)
+		}
+		if v == sha256 {
```

Comment:
this was the problem, this would default to cr and then fo into an infinite loop in trying to resolve the config